crates/fuel-core-nats/src/nats.rs

```diff
@@ -37,7 +37,12 @@ impl NatsConnection {
             )
         }
         // Publish
-        let ack_future = self.jetstream.publish(subject, payload).await?;
+        let mut headers_map = HeaderMap::new();
+        headers_map.append(NATS_MESSAGE_ID, subject.to_string().as_str());
```

Comment:
what is this `HeaderMap` for?

Reply:
The headermap sets some dedup headers which is btw. achieved with .message_id(...) which I have now switched to here. Sorry, I did not see the method has a message_id.